### ISSUE TITLE
torchaudio.transforms.TimeMasking is inline

### ISSUE THREAD

**Issue by seacj** [pytorch/pytorch#57268]
## 🐛 Bug

The function torchaudio.transforms.MelSpectrogram transforms the Tensor inline without expectation.

## To Reproduce

import matplotlib.pyplot as plt
import librosa
import torchaudio
import torch
fmask = torchaudio.transforms.TimeMasking(time_mask_param=5)

BN = 2
max_frames = 200
data = torch.rand((BN, max_frames*160+240))
mt = torchaudio.transforms.MelSpectrogram(sample_rate=16000, n_fft=512, win_length=400, hop_length=160,
                                                 window_fn=torch.hamming_window, n_mels=64)
spec = mt(data)

def plot_spectrogram(spec, title=None, ylabel='freq_bin', aspect='auto', xmax=None):
    fig, axs = plt.subplots(1, 1)
    axs.set_title(title or 'Spectrogram (db)')
    axs.set_ylabel(ylabel)
    axs.set_xlabel('frame')
    im = axs.imshow(librosa.power_to_db(spec), origin='lower', aspect=aspect)
    if xmax:
        axs.set_xlim((0, xmax))
        fig.colorbar(im, ax=axs)
        plt.show(block=False)

spec = mt(data)

fmask(spec) # expect `mask_spec = fmask(spec)` or `fmask(spec, inline=False)`
print((spec==mt(data)).all()) # False (but  True is expected)

## Expected behavior

expect `mask_spec = fmask(spec)` or `fmask(spec, inline=False)`

## Environment
Collecting environment information...
PyTorch version: 1.8.1+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 16.04.7 LTS (x86_64)
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
Clang version: Could not collect
CMake version: version 3.5.1

Python version: 3.8 (64-bit runtime)
Is CUDA available: True

Versions of relevant libraries:
[pip3] numpy==1.20.2
[pip3] pytorch-metric-learning==0.9.97
[pip3] torch==1.8.1
[pip3] torch-scatter==2.0.6
[pip3] torchaudio==0.8.1
[pip3] torchlars==0.1.2
[pip3] torchvision==0.7.0
[conda] blas                      1.0                         mkl    defaults
[conda] cudatoolkit               10.1.243             h6bb024c_0    defaults
[conda] mkl                       2020.2                      256    defaults
[conda] mkl-service               2.3.0            py38he904b0f_0    defaults
[conda] mkl_fft                   1.2.0            py38h23d657b_0    defaults
[conda] mkl_random                1.1.1            py38h0573a6f_0    defaults
[conda] numpy                     1.19.2           py38h54aff64_0    defaults
[conda] numpy-base                1.19.2           py38hfa32c7d_0    defaults
[conda] pytorch                   1.6.0           py3.8_cuda10.1.243_cudnn7.6.3_0    pytorch
[conda] pytorch-metric-learning   0.9.97                   pypi_0    pypi
[conda] torch-scatter             2.0.6                    pypi_0    pypi
[conda] torchaudio                0.6.0                      py38    pytorch
[conda] torchlars                 0.1.2                    pypi_0    pypi
[conda] torchvision               0.7.0                py38_cu101    pytorch

**Comment by gchanan** [comment]
do you mean inplace?
